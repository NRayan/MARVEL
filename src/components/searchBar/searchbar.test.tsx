// import { render } from "@testing-library/react-native";
import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { RenderWithTheme } from "../../tests";
import { SearchBar } from "./";

describe("SearchBar", () => {

	it("Should Render correctly", () => {

		const tree = render(
			<RenderWithTheme>
				<SearchBar onChangeText={() => null} />
			</RenderWithTheme>
		).toJSON();

		expect(tree).toMatchSnapshot();

	});

	it("Should display clear button after typing", () => {

		const { getByPlaceholderText, queryByTestId } = render(
			<RenderWithTheme>
				<SearchBar onChangeText={() => null} />
			</RenderWithTheme>
		);

		expect(queryByTestId("clear-button")).toBeFalsy();

		const textInput = getByPlaceholderText("search...");
		fireEvent.changeText(textInput, "Iron Man");

		expect(queryByTestId("clear-button")).toBeTruthy();

	});
	it("Should clear de text after pressing the clear button", () => {

		const { getByPlaceholderText, queryByTestId } = render(
			<RenderWithTheme>
				<SearchBar onChangeText={() => null} />
			</RenderWithTheme>
		);

		const searchtext = "Iron Man";

		const textInput = getByPlaceholderText("search...");
		fireEvent.changeText(textInput, searchtext);

		fireEvent.press(queryByTestId("clear-button"));

		expect(textInput.props.value).toBe("");

	});

});

