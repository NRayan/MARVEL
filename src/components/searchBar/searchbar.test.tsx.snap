// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should Render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#21263e",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 15,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 15,
        "paddingBottom": 0,
        "paddingLeft": 10,
        "paddingRight": 0,
        "paddingTop": 0,
        "width": "100%",
      },
    ]
  }
>
  <Text />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="search..."
    placeholderTextColor="rgba(255,255,255,0.4)"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "marginBottom": 15,
          "marginLeft": 10,
          "marginRight": 0,
          "marginTop": 15,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
